### PR TITLE
docs: add documentation for tagging releases

### PR DESCRIPTION
**Pull request by rebello95**
Signed-off-by: Michael Rebello <me@michaelrebello.com>

### CODE REVIEW

File: docs/root/development/releasing/releasing.rst [hunk 1]
```diff
@@ -0,0 +1,40 @@
+.. _releasing_instructions:
+
+Releasing
+=========
+
+Prepare for release
+-------------------
+
+When a new `release <https://github.com/lyft/envoy-mobile/releases>`_ is ready to be tagged,
+the following workflow should be followed to create and publish a new Envoy Mobile release.
```
Comment (junr03) [inline docs/root/development/releasing/releasing.rst]: reword as:

> The following workflow should be followed to create and publish a new Envoy Mobile `release <https://github.com/lyft/envoy-mobile/releases>`_.

Comment (rebello95) [reply docs/root/development/releasing/releasing.rst]: SGTM, updated

File: docs/root/development/releasing/releasing.rst [hunk 2]
```diff
@@ -0,0 +1,40 @@
+.. _releasing_instructions:
+
+Releasing
+=========
+
+Prepare for release
+-------------------
+
+The following workflow should be followed to create and publish a new Envoy Mobile
```
Comment (junr03) [inline docs/root/development/releasing/releasing.rst]: sorry, this is silly, but could you put this sentence between Releasing and prepare for release? It applies to both subsections.

Comment (rebello95) [reply docs/root/development/releasing/releasing.rst]: done.

**Pull request merged**